TA_Scheduler/views/account/create.py

```diff
@@ -1,7 +1,45 @@
 from django.shortcuts import render
 from django.views import View
+from TA_Scheduler.models import Account
+from TA_Scheduler.utilities.AccountUtil import AccountUtil
 
 
 class CreateAccount(View):
     def get(self, request):
-        return render(request, "account/create.html")
+        return render(request, "account/create.html", {"message": " "})
+
+    def post(self, request):
+        username = request.POST["username"]
+        if AccountUtil.getAccountByUsername(username) is None:
+            return render(
+                request,
+                "account/create.html",
+                {"message": "username '" + username + "' is already in use"},
+            )
+        else:
+            usertype = int(request.POST["authority"])
+            if usertype == 1:
+                AccountUtil.createAdminAccount(username, request.POST["password"])
+                return render(
+                    request,
+                    "account/create.html",
+                    {"message": "Admin account '" + username + "' created"},
+                )
+            elif usertype == 2:
+                AccountUtil.createInstructorAccount(username, request.POST["password"])
+                return render(
+                    request,
+                    "account/create.html",
+                    {"message": "Instructor account '" + username + "' created"},
+                )
+            elif usertype == 3:
+                AccountUtil.createTAAccount(username, request.POST["password"])
+                return render(
+                    request,
+                    "account/create.html",
+                    {"message": "TA account '" + username + "' created"},
+                )
+            else:
+                return render(
+                    request, "account/create.html", {"message": "enter user type"}
+                )
```

Comment:
We don't use authority anymore, so I would recommend switching to using groups
   
```python
if request.user.is_anonymous:
    return redirect("/", {"error": "User is not authorized to create a course"})
elif request.user.groups.filter(name="admin).exists():
    #Something if admin
elif request.user.groups.filter(name="instructor").exists():
    #Something if instructor
elif request.user.groups.filter(name="ta").exists():
    #Something if ta
else:
   return redirect("/", {"error": "User is not authorized to create a course"})
```

Reply:
usertype is something returned from the form (the type of user being created).  An if/elif/else statement like you suggest wouldn't really work as a replacement for the one I have in place.

Reply:
oh yeah mb. i read it wrong